libr/anal/var.c

```diff
@@ -114,16 +110,43 @@ R_API bool r_anal_var_rebase(RAnal *a, RAnalFunction *fcn, ut64 diff) {
 	return true;
 }
 
-R_API bool r_anal_var_add(RAnal *a, ut64 addr, int scope, int delta, char kind, R_NULLABLE const char *type, int size, bool isarg, R_NONNULL const char *name) {
-	r_return_val_if_fail (a && name, false);
+// If the type of var is a struct,
+// remove all other vars that are overlapped by var and are at the offset of one of its struct members
+static void shadow_var_struct_members(RAnalFunction *fcn, RAnalVar *var) {
+	Sdb *TDB = fcn->anal->sdb_types;
+	const char *type_kind = sdb_const_get (TDB, var->type, 0);
+	if (type_kind && r_str_startswith (type_kind, "struct")) {
+		char *field;
+		int field_n;
```

Comment:
use size_t for loops

Reply:
It's ok here since it is passed to `sdb_array_get()`